### PR TITLE
♻️ Mobile | Cleanup LoginViewModel and Auth service

### PR DESCRIPTION
**Pull request by AntPolkanov**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ #731

> 2. What was changed?

✏️ Removed duplicate property `HasCachedAccount` from the `UserService`
Removed code which reads data from `IdentityToken`
Fixed error which popups up when the user closes the browser without logging-in.

> 3. Did you do pair or mob programming?

✏️ No

### CODE REVIEW

File: src/MobileUI/Services/IAuthenticationService.cs [hunk 1]
```diff
@@ -130,31 +125,7 @@ private ApiStatus SetLoggedInState(AuthResult loginResult)
             try
             {
                 Preferences.Set(nameof(HasCachedAccount), true);
-
-                var tokenHandler = new JwtSecurityTokenHandler();
-
-                var jwToken = tokenHandler.ReadJwtToken(loginResult.IdentityToken);
```
Comment (AntPolkanov) [inline src/MobileUI/Services/IAuthenticationService.cs:136]: Not used anywhere.

File: src/MobileUI/Services/IUserService.cs [hunk 2]
```diff
@@ -12,7 +12,6 @@ public interface IUserService
     IObservable<int> MyBalanceObservable();
     IObservable<string> MyQrCodeObservable();
     IObservable<int> MyAllTimeRankObservable();
-    bool HasCachedAccount { get; }
```
Comment (AntPolkanov) [inline src/MobileUI/Services/IUserService.cs:15]: Duplicate.

File: src/MobileUI/ViewModels/LoginPageViewModel.cs [hunk 3]
```diff
@@ -35,35 +33,27 @@ private async Task LoginTapped()
         LoginButtonEnabled = false;
         bool enableButtonAfterLogin = true;
 
-        ApiStatus status;
-        try
-        {
-            status = await _authService.SignInAsync();
-        }
-        catch (Exception exception)
```
Comment (AntPolkanov) [inline src/MobileUI/ViewModels/LoginPageViewModel.cs:43]: Redundant as `SignInAsync()` catches all exceptions

File: src/MobileUI/ViewModels/LoginPageViewModel.cs [hunk 4]
```diff
@@ -82,43 +72,40 @@ private async static Task WaitForWindowClose()
 
     public async Task Refresh()
     {
+        if (!_authService.HasCachedAccount)
```
Comment (AntPolkanov) [inline src/MobileUI/ViewModels/LoginPageViewModel.cs:75]: Inverted `if`

**Review by matt-goldman — APPROVED** [review]
LGTM 🚀

**Pull request merged**